index.js

```diff
@@ -127,25 +128,34 @@ Parcelify.prototype.processParcels = function( browserifyInstance, options, call
 			process.nextTick( function() {
 				async.series( [ function( nextSeries ) {
 					// fire package events for any new packages
-					_.each( packagesThatWereCreated, function( thisPackage ) {
-						var isParcel = thisPackage.isParcel;
+					var pipelinePackages = _.map( packagesThatWereCreated, function( thisPackage ) {
+						return function (nextPackage) {
+							var isParcel = thisPackage.isParcel;
+
+							log.verbose( 'Created new ' + ( isParcel ? 'parcel' : 'package' ) + ' ' + thisPackage.path + ' with id ' + thisPackage.id );
 
-						log.verbose( 'Created new ' + ( isParcel ? 'parcel' : 'package' ) + ' ' + thisPackage.path + ' with id ' + thisPackage.id );
+							existingPackages[ thisPackage.id ] = thisPackage;
+							if( isParcel ) {
+								_this._setupParcelEventRelays( thisPackage );
+								parcelsThatWereCreated.push( thisPackage );
+							}
 
-						existingPackages[ thisPackage.id ] = thisPackage;
-						if( isParcel ) {
-							_this._setupParcelEventRelays( thisPackage );
-							parcelsThatWereCreated.push( thisPackage );
-						}
+							thisPackage.on( 'error', function( err ) {
+								_this.emit( 'error', err );
+							} );
 
-						thisPackage.on( 'error', function( err ) {
-							_this.emit( 'error', err );
-						} );
+							var pipelineProcessors = _.map( _this.preProcessors, function (p) {
+								return async.apply( p, thisPackage);
+							});
 
-						_this.emit( 'packageCreated', thisPackage );
+							async.parallel(pipelineProcessors, nextPackage);
+
+							_this.emit( 'packageCreated', thisPackage );
+						};
 					} );
 
-					nextSeries();
+					async.parallel(pipelinePackages, nextSeries);
```

Comment:
Here is the magic, we will not move to the next series function until all the processors are finished. We still emit `packageCreate` event